Assets/MixedRealityToolkit/Utilities/AssetScriptReferenceRetargeter.cs

```diff
@@ -343,10 +364,164 @@ private static bool IsYamlFile(string filePath)
+            if (uapMetaFile == null)
+            {
+                Debug.LogError("Could not find sample UAP dll.meta file");
+            }
+            FileInfo standaloneMetaFile = assetDirectory.GetFiles("*sampleStandaloneDllMeta.txt", SearchOption.AllDirectories).FirstOrDefault();
+            if (standaloneMetaFile == null)
+            {
+                Debug.LogError("Could not find sample Standalone dll.meta file");
+            }
+
+            string[] metaFileContent = File.ReadAllLines(editorMetaFile.FullName);
+
+            foreach (FileInfo dllFile in new DirectoryInfo(Application.dataPath.Replace("Assets", "NuGet/Plugins")).GetFiles("*", SearchOption.AllDirectories))
+            {
+                if (dllFile.Extension.Equals(".meta"))
+                {
+                    string dllFileName = dllFile.Name.Remove(dllFile.Name.Length - 5);
+                    if (dllGuids.ContainsKey(dllFileName))
+                    {
+                        metaFileContent[1] = string.Format("guid: {0}", dllGuids[dllFileName]);
+                        File.WriteAllLines(dllFile.FullName, metaFileContent);
+                    }
+                }
+            }
+
+            DirectoryInfo[] directories = new DirectoryInfo(Application.dataPath.Replace("Assets", "NuGet/Plugins")).GetDirectories("*", SearchOption.AllDirectories);
+            string dllGuid = string.Empty;
+            bool isUAP = false;
+            foreach (DirectoryInfo directory in directories)
+            {
+                isUAP = false;
+                if (!directory.Name.Equals("Editor"))
+                {
+                    if (directory.Name.Contains("Standalone") || directory.Parent.Name.Contains("Standalone"))
+                    {
+                        metaFileContent = File.ReadAllLines(standaloneMetaFile.FullName);
+                    }
+                    else if (directory.Name.Contains("UAP") || directory.Parent.Name.Contains("UAP"))
+                    {
+                        metaFileContent = File.ReadAllLines(uapMetaFile.FullName);
+                        isUAP = true;
+                    }
+
+                    FileInfo[] files = directory.GetFiles("*.dll", SearchOption.TopDirectoryOnly);
+                    string metaFilePath = string.Empty;
+                    foreach (FileInfo file in files)
+                    {
+                        //Editor is guid + 1
+                        //Standalone is guid + 2
+                        //UAP is guid + 3
+                        dllGuid = dllGuids[file.Name];
+                        dllGuid = CycleGuidForward(dllGuid);
+                        if (isUAP)
+                        {
+                            dllGuid = CycleGuidForward(dllGuid);
+                        }
+
+                        if (dllGuids.ContainsKey(file.Name))
+                        {
+                            metaFilePath = string.Format("{0}.meta", file.FullName);
```

Comment:
> string.Format("{0}.meta", file.FullName); [](start = 42, length = 42)

Could use the new syntax and also inline the declaration of the field (remove from line 476